### ISSUE TITLE
msgfmt is no longer installed

### ISSUE THREAD

**Issue by Arthri** [git-for-windows/git#4903]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.44.0.windows.1
cpu: x86_64
built from commit: ad0bbfffa543db6979717be96df630d3e5741331
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19045.4046]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"
> type "C:\Program Files (x86)\Git\etc\install-options.txt"
> type "%USERPROFILE%\AppData\Local\Programs\Git\etc\install-options.txt"
> type "$env:USERPROFILE\AppData\Local\Programs\Git\etc\install-options.txt"
$ cat /etc/install-options.txt

Editor Option: VisualStudioCode
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: LFOnly
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Enabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
sh -c "where msgfmt"
```
 - What did you expect to occur after running these commands?

I expected the command to output C:\Program Files\Git\usr\bin\msgfmt.exe like it did on Bash 5.2.21(1)-release

 - What actually happened instead?

INFO: Could not find files for the given pattern(s).

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

https://github.com/Arthri/demo-git-where-msgfmt

https://github.com/Arthri/demo-git-where-msgfmt/actions/runs/8610401685/job/23595857738

**Comment by dscho** [comment]
The purpose of [Git for Windows](https://git-for-windows.github.io/) is really to bring *Git* to Windows.

While that implies that we have to ship quite a few Unix-like tools, it does not mean that we will include such tools unless they are needed for Git's own operations, or at least for historical reasons (needs of active contributors are sometimes a factor in deciding whether to include a tool or not).

<details><summary>So why was msgfmt.exe included in Git for Windows <v2.44.0?</summary>

Let's first get one thing out of the way: `msgfmt.exe` is in no way, shape or form required to run Git. Therefore, there was _no_ good reason why this tool was shipped with Git for Windows in the first place.

So why was it?

To support diffing, say, Word documents, Git for Windows includes the `astextplain` script that requires the `file` executable to be present and in working order. `file`, in turn depends on `liblzma`, which listed `gettext` as _runtime dependency_ until [that mistake was fixed](https://github.com/msys2/MSYS2-packages/commit/d7c37ffdadd8513f84a16b1565a5f9df75e37af2).

In other words, `msgfmt.exe` was shipped as part of the `gettext` package that was a (needless) dependency of `liblzma`, which in turn is a dependency of `file`, which is required by `astextplain` that is included in the `mingw-w64-git-extra` package. And when that needless dependency was removed, so was `msgfmt.exe`.

</details>

Having said that, what you really are looking for is [MSYS2](https://msys2.github.io/). Git for Windows leverages MSYS2 and ships with a subset of its files. MSYS2 even sports a package management system called "pacman" to install more tools (including Git...), so why don't you give it a whirl?
